packages/superset-ui-encodable/src/encoders/Encoder.ts

```diff
@@ -92,6 +95,55 @@ export default class Encoder<Config extends EncodingConfig> {
     return Array.from(new Set(fields));
   }
 
+  private createLegendItemsFactory(field: string) {
+    const channelEncoders = flatMap(
+      this.getChannelEncoders().filter(e => isNotArray(e) && isValueDef(e.definition)),
+    ).concat(this.legends[field]);
+
+    return (domain: ChannelInput[]) =>
+      domain.map((input: ChannelInput) => ({
+        input,
+        output: channelEncoders.reduce(
+          (prev: Partial<{ [k in keyof Config]: Config[k]['1'] }>, curr) => {
+            const map = prev;
+            map[curr.name as keyof Config] = curr.encodeValue(`${input}`) as Value;
```

Comment:
this works even if `ChannelInput` is not a `string`? why coerce to one? (could add comment)

Reply:
not sure why I did that. removed now.